### PR TITLE
adds activity log triggers for project features

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/10809

## Testing
**URL to test:** 

local! since you need to access the hasura console to test. this one is a little tricky to test since these changes are only reflected in the `moped_activity_log` table, so i've included detailed testing instructions below.

**Steps to test:**

this pr adds activity log tracking for project features. here are the tables and how to test them!

`feature_drawn_lines`
- create and save a project line feature component (e.g., bike lane - advisory)
- edit the component and click the map tool in the top right to add a line to the existing line feature
- reopen the component and delete the drawn line
- delete the component
- the `moped_activity_log` table in hasura console should show rows(s) for the actions associated with `feature_drawn_lines` and `moped_proj_components`

`feature_drawn_points`
- create and save a project point feature component (e.g., bike box)
- edit the component and click the map tool in the top right to add a point to the existing point feature
- reopen the component and delete the drawn point
- delete the component
- the `moped_activity_log` table in hasura console should show rows(s) for the actions associated with `feature_drawn_points` and `moped_proj_components`

`feature_intersections`
- create and save an intersection component (e.g., intersection - improvement)
- edit the component to add or remove an intersection point
- delete the intersection component
- the `moped_activity_log` table in hasura console should show rows(s) for the actions associated with `feature_intersections` and `moped_proj_components`

`feature_signals`
- create and save a signal component (e.g., signal - phb)
- delete the signal component
- the `moped_activity_log` table in hasura console should show rows(s) for the actions associated with `feature_signals` and `moped_proj_components`

`feature_street_segments`
- create and save a street segment component (e.g., access control - driveway closure)
- edit the component to add or remove a street segment
- delete the street segment component
- the `moped_activity_log` table in hasura console should show rows(s) for the actions associated with `feature_street_segments` and `moped_proj_components`

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-database/metadata/tables.yaml [hunk 1]
```diff
@@ -153,6 +153,53 @@
...
+                  "record_id": {{ $body.event.data.new.id }},
+                  "record_type":  {{ $body.table.name }},
+                  "activity_id": {{ $body.id }},
+                  "record_data": {"event": {{ $body.event }}},
+                  "description": [{"newSchema": "true"}],
+                  "operation_type": {{ $body.event.op }},
+                  "updated_by_user_id": {{ $session_variables?['x-hasura-user-db-id'] ?? 1}}
+                }
+              }
+            }
```
Comment (johnclary) [inline moped-database/metadata/tables.yaml:191]: Just pointing out that because these records do not have a `project_id` column, we unfortunately cannot update the project's modified date, nor can we set the `record_project_id` column in the activity log. 

we'll leave that challenge for another today—i'm just glad to be tracking changes to these records.

also, i noticed that the `moped_activity_log.record_project_id` column defaults to `0`, which i think should be changed to `null`. i'm going to open a separate issue for that b/c i believe we can also clean up a few other columns.

TLDR—looks good!

**Review by johnclary — APPROVED** [review]
thanks @tillyw! you nailed it!

one observation for the group: if a user creates a component with a lot of features, e.g. they select 50 street segments, that will result in 50 activity events firing, one per feature. i really think this is ok—the biggest risk I think we're running is forgetting to switch off these triggers if we need to make a massive bulk update.

**Review by chiaberry — APPROVED** [review]
Thanks for the detailed testing notes!

File: moped-database/metadata/tables.yaml (hunk 1 above)
Comment (mddilley) [reply moped-database/metadata/tables.yaml:191]: Thanks for that context, y'all! I was going to ask about the missing value in the `record_project_id` column, but, now, I understand why. 🙏

**Review by mddilley — APPROVED** [review]
This looks amazing! So cool to see these updates in the activity log table, and thanks for mentioning to look in the Hasura console. I tested each type of feature and saw the expected entries. 🚢 🚢 🚢
